includes/sanitizers/class-amp-script-sanitizer.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Class AMP_Script_Sanitizer
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Script_Sanitizer
+ *
+ * @since 1.0
+ */
+class AMP_Script_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Sanitize noscript elements.
+	 *
+	 * Eventually this should also handle script elements, if there is a known AMP equivalent.
+	 * If nothing is done with script elements, the whitelist sanitizer will deal with them ultimately.
+	 *
+	 * @todo Eventually this try to automatically convert script tags to AMP when they are recognized. See <https://github.com/Automattic/amp-wp/issues/1032>.
+	 * @todo When a script has an adjacent noscript, consider removing the script here to prevent validation error later. See <https://github.com/Automattic/amp-wp/issues/1213>.
+	 *
+	 * @since 1.0
+	 */
+	public function sanitize() {
+		$noscripts = $this->dom->getElementsByTagName( 'noscript' );
+
+		for ( $i = $noscripts->length - 1; $i >= 0; $i-- ) {
+			$noscript = $noscripts[ $i ];
+
+			// Skip AMP boilerplate.
+			if ( $noscript->firstChild instanceof DOMElement && $noscript->firstChild->hasAttribute( 'amp-boilerplate' ) ) {
+				continue;
+			}
+
+			$fragment = $this->dom->createDocumentFragment();
+			$fragment->appendChild( $this->dom->createComment( 'noscript' ) );
+			while ( $noscript->firstChild ) {
+				$fragment->appendChild( $noscript->firstChild );
+			}
+			$fragment->appendChild( $this->dom->createComment( '/noscript' ) );
```

Comment:
It's nice how this outputs the comment `<!--noscript-->`. This should help with debugging if needed, like on [WordPress.org support topics](https://wordpress.org/support/plugin/amp).